### PR TITLE
bpo-46306: simplify `CodeType` attribute access in `doctest.py`

### PR DESCRIPTION
**Pull request by sobolevn**
https://bugs.python.org/issue46306

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2022-01-08-13-53-25.bpo-46306.1_es8z.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Assume that :class:`types.CodeType` always has ``co_firstlineno`` prop in
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2022-01-08-13-53-25.bpo-46306.1_es8z.rst]:
```suggestion
Assume that :class:`types.CodeType` always has :attr:`types.CodeType.co_firstlineno` in
```

**Review by kumaraditya303 — APPROVED** [review]
LGTM

File: Misc/NEWS.d/next/Library/2022-01-08-13-53-25.bpo-46306.1_es8z.rst (hunk 1 above)
Comment (sobolevn) [reply Misc/NEWS.d/next/Library/2022-01-08-13-53-25.bpo-46306.1_es8z.rst]: @kumaraditya303 is `:attr:` context specific? Or does it need the full path like `types.CodeType.co_firstlineno`?

Comment (kumaraditya303) [reply Misc/NEWS.d/next/Library/2022-01-08-13-53-25.bpo-46306.1_es8z.rst]: Yeah full path would be better, Updated suggestion!

Comment (sobolevn) [reply Misc/NEWS.d/next/Library/2022-01-08-13-53-25.bpo-46306.1_es8z.rst]: Thank you! 👍

**Review by ericvsmith — APPROVED** [review]
This looks good to me.

**Comment by ericvsmith** [comment]
Does this need backporting?

**Comment by sobolevn** [comment]
@ericvsmith this was 18 years old before I've changed it 🤔 
Probably it can be an improvement only.

**Comment by ericvsmith** [comment]
Heh. Okay.

**Pull request merged**